crates/node/service/src/service/util.rs

```diff
@@ -0,0 +1,34 @@
+//! Utilities for the rollup node service, internal to the crate.
+
+/// Spawns a set of parallel actors in a [JoinSet], and cancels all actors if any of them fail. The
+/// type of the error in the [NodeActor]s is erased to avoid having to specify a common error type
+/// between actors.
+///
+/// [JoinSet]: tokio::task::JoinSet
+/// [NodeActor]: crate::NodeActor
+macro_rules! spawn_and_wait {
+    ($cancellation:expr, actors = [$($actor:expr$(,)?)*]) => {
+        let mut task_handles = tokio::task::JoinSet::new();
+
+        $(
+            task_handles.spawn(async move {
+                if let Err(e) = $actor.start().await {
+                    // TODO: Bubble up generic error.
```

Comment:
Shouldn't this just return the error so the `join_next` call below will pick it up and gracefully shutdown?

Reply:
Yes it should. Adding in a type erased error type in a follow up PR, goal here is to allow the actors to have separate error defs but exist in the same joinset.